sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasingSuite.scala

```diff
@@ -493,6 +491,156 @@ class NestedColumnAliasingSuite extends SchemaPruningTest {
     comparePlans(optimized3, expected3)
   }
 
+  test("Nested field pruning for Window") {
+    val spec = windowSpec($"address" :: Nil, $"id".asc :: Nil, UnspecifiedFrame)
+    val winExpr = windowExpr(RowNumber(), spec)
+
+    val query1 = contact
+      .select($"name.first", winExpr.as('window))
+      .analyze
+    val optimized1 = Optimize.execute(query1)
+    val expected1 = contact
+      .select($"name.first", $"address", $"id")
+      .window(Seq(winExpr.as("window")), Seq($"address"), Seq($"id".asc))
+      .select($"first", $"window")
+      .analyze
+    comparePlans(optimized1, expected1)
```

Comment:
Is this test pruning nested column? I don't see generated alias here.

Reply:
It is pruning but not generating aliases. 
https://github.com/apache/spark/pull/28898/files/a0998ae4166efff59b62971e98945e201809fa15#diff-d87f0060a604ac3e7149bd108fd548a5R505
This line suggests it only select `first` instead of the whole column `name`.

Reply:
But I ran this test in current master and it passed.

Reply:
Exactly, this isn’t improved in this pr as it always works. Not sure whether we want to remove this test.

Reply:
It passed without pruning through window. The query selects `name.first` on top of the relation. So I am not sure why you include it here...

Reply:
Good point. Removed such test.